lib/web_ui/test/matchers.dart

```diff
@@ -303,6 +303,9 @@ String canonicalizeHtml(String htmlContent,
 
     if (mode != HtmlComparisonMode.noAttributes) {
       original.attributes.forEach((dynamic name, String value) {
+        if (name is! String) {
```

Comment:
is check at runtime needed? assert?

Reply:
Yes, only an `if` statement causes the compiler/analyzer to promote the type. There's no performance penalty here because this is test code.